32.md

```diff
@@ -0,0 +1,172 @@
+- `confidence` may have a value of 0 to 1. This indicates the certainty which the author has about their rating.
+- `context` may be a comma-separated list of urls (including NIP-21 urls) indicating other context that should be considered when interpreting labels.
+
+Content
+-------
+
+Labels should be short, meaningful strings. Longer discussions, such as for a review, or an
+explanation of why something was labeled the way it was, should go in the event's `content` field.
+
+Example events
+--------------
+
+A report that an event contains nudity. Note that NIP 56 is preferred for reporting content
+to clients, while labels are recommended for supporting distributed content moderation use
+cases.
+
+```json
+{
+  "kind": 1985,
+  "tags": [
+    ["L", "report"],
+    ["l", "nudity", "report"],
+    ["e", <id>, <relay_url>]
+  ],
+  "content": "",
+  ...
+}
+```
+
+A single event can apply multiple labels to multiple targets to support mass-tagging. Multiple
+namespaces may be used at the same time.
+
+```json
+{
+  "kind": 1985,
+  "tags": [
+    ["e", <id>, <relay_url>],
+    ["p", <id>, <relay_url>],
+    ["t", "chickens"],
+    ["L", "#t"]
+    ["L", "ugc"]
+    ["L", "com.example.labels"]
+    ["l", "chickens", "#t"],
+    ["l", "user generated content", "ugc"],
+    ["l", "permaculture", "com.example.labels"],
+    ["l", "permies", "com.example.labels"],
+    ["l", "farming", "com.example.labels"],
+  ],
+  "content": "",
+  ...
+}
+```
+
+A suggestion that multiple pubkeys be associated with the `permies` topic.
+
+```json
+{
+  "kind": 1985,
+  "tags": [
+    ["L", "#t"],
+    ["l", "permies", "#t"],
+    ["p", <pubkey1>, <relay_url>],
+    ["p", <pubkey2>, <relay_url>]
+  ],
+  "content": "",
+  ...
+}
+```
+
+A review of a relay, as relates to certain topics, including additional dimensions. The author
+is indicating here that `relay_url` is related to the bitcoin topic, but they're not very sure
+that's the case.
+
+```json
+{
+  "kind": 1985,
+  "tags": [
+    ["L", "#t"],
+    ["l", "bitcoin", "#t", "{\"quality\": 0.7, \"confidence\": 0.2}"],
+    ["r", <relay_url>]
+  ],
+  "content": "I think this relay is mostly just bitcoiners.",
+  ...
+}
+```
+
+A plain review of a relay.
+
+```json
+{
+  "kind": 1985,
+  "tags": [
+    ["L", "social.coracle.ontology"],
+    ["l", "review", "social.coracle.ontology", "{\"quality\": 0.1}"],
+    ["r", <relay_url>]
+  ],
+  "content": "This relay is full of mean people.",
+  ...
+}
+```
+
+A more abstract use case: defining an edge in a graph structure, in this case identifying
+a lightning channel that is open between two pubkeys. This just demonstrates the flexibility
+this spec provides for overlaying structured metadata on top of nostr.
+
+```json
+{
+  "kind": 1985,
+  "tags": [
+    ["L", "my-lightning-nomenclature"],
+    ["l", "channel", "my-lightning-nomenclature"],
+    ["p", <pubkey1>, <relay_url>],
+    ["p", <pubkey2>, <relay_url>]
+  ],
+  "content": "<channel_id>",
+  ...
+}
+```
```

Comment:
Allowing labels directly on event kinds other than 1985 means that "L" and "l" will have a reserved meaning, no matter the kind. So, for example, nobody may use `l` meaning "language" on a kind xx.

Maybe in your example its best to create a kind 1 and also a kind 1985 event.

Reply:
From the NIP:

> `l` and `L` tags MAY be added to other event kinds to support self-reporting. For events with a kind other than 1985, labels refer to the event itself.

If someone wants to designate language, a new tag isn't necessary, they can do `["l", "en", "ISO 639-1"]`.

Reply:
"Language" was just an example, it could mean anything in another kind context. It was just a side note for you to consider. Not against it I'm neutral on this specific topic.

Reply:
Just added an example for self-tagging.

Reply:
> "Language" was just an example

Gotcha, I think this is ok, but if you see a bigger potential problem let me know. Do we already use "language" tags?

Reply:
> Do we already use "language" tags?

There is [this PR](https://github.com/nostr-protocol/nips/pull/182). There's other open issue with `l` meaning "tree (l)evel/depth". But nothing using the `l` tag that I know became a NIP yet, so no problem.